### ISSUE TITLE
cmake 3.20 doesn't work when using gcc2 to compile projects

### ISSUE THREAD

**Issue by augiedoggie** [haikuports/haikuports#5835]
3.19 works fine

Example output from `haikuporter tidy` on Haiku B2:

```
~/haikuports> haikuporter tidy
Checking if any dependency-infos need to be updated ...
Looking for stale dependency-infos ...
----------------------------------------------------------------------
app-text::tidy-5.1.25
        /boot/home/haikuports/app-text/htmltidy/tidy-5.1.25.recipe
----------------------------------------------------------------------
Skipping download of source for 5.1.25.tar.gz
Validating checksum of 5.1.25.tar.gz
Unpacking source of 5.1.25.tar.gz
hint: Using 'master' as the name for the initial branch. This default branch name
hint: is subject to change. To configure the initial branch name to use in all
hint: of your new repositories, which will suppress this warning, call:
hint: 
hint:   git config --global init.defaultBranch <name>
hint: 
hint: Names commonly chosen instead of 'master' are 'main', 'trunk' and
hint: 'development'. The just-created branch can be renamed via this command:
hint: 
hint:   git branch -m <name>
Initialized empty Git repository in /boot/home/haikuports/app-text/htmltidy/work-5.1.25/sources/tidy-html5-5.1.25/.git/
Applying patchset "/boot/home/haikuports/app-text/htmltidy/patches/tidy-5.1.25.patchset" ...
Applying: Port to Haiku.
chroot has these packages active:
        /boot/system/packages/bash-5.1-1-x86_gcc2.hpkg
        /boot/system/packages/binutils-2.17_2016_07_24-4-x86_gcc2.hpkg
        /boot/system/packages/bzip2-1.0.8-1-x86_gcc2.hpkg
        /boot/system/packages/bzip2_x86-1.0.8-1-x86_gcc2.hpkg
        /boot/system/packages/bzip2_x86_devel-1.0.8-1-x86_gcc2.hpkg
        /boot/system/packages/ca_root_certificates-2021_01_19-1-any.hpkg
        /boot/system/packages/cmake_x86-3.20.0-1-x86_gcc2.hpkg
        /boot/system/packages/coreutils-8.24-2-x86_gcc2.hpkg
        /boot/system/packages/curl-7.74.0-1-x86_gcc2.hpkg
        /boot/system/packages/curl_x86-7.74.0-1-x86_gcc2.hpkg
        /boot/system/packages/dav1d_x86-0.8.2-1-x86_gcc2.hpkg
        /boot/system/packages/enca_x86-1.19-2-x86_gcc2.hpkg
        /boot/system/packages/expat-2.2.10-1-x86_gcc2.hpkg
        /boot/system/packages/expat_x86-2.2.10-1-x86_gcc2.hpkg
        /boot/system/packages/fdk_aac_x86-2.0.1-1-x86_gcc2.hpkg
        /boot/system/packages/ffmpeg-4.2.2-9-x86_gcc2.hpkg
        /boot/system/packages/ffmpeg_x86-4.2.2-9-x86_gcc2.hpkg
        /boot/system/packages/file-5.39-2-x86_gcc2.hpkg
        /boot/system/packages/file_data-5.39-2-any.hpkg
        /boot/system/packages/fontconfig-2.13.92-1-x86_gcc2.hpkg
        /boot/system/packages/fontconfig_x86-2.13.92-1-x86_gcc2.hpkg
        /boot/system/packages/freetype-2.10.4-2-x86_gcc2.hpkg
        /boot/system/packages/freetype_bootstrap_x86-2.10.4-1-x86_gcc2.hpkg
        /boot/system/packages/freetype_x86-2.10.4-2-x86_gcc2.hpkg
        /boot/system/packages/fribidi_x86-1.0.10-1-x86_gcc2.hpkg
        /boot/system/packages/game_music_emu_x86-0.6.3-1-x86_gcc2.hpkg
        /boot/system/packages/gcc-2.95.3_2017_07_20-2-x86_gcc2.hpkg
        /boot/system/packages/gcc_x86_syslibs-8.3.0_2019_05_24-9-x86_gcc2.hpkg
        /boot/system/packages/gettext_libintl-0.19.8.1-7-x86_gcc2.hpkg
        /boot/system/packages/gettext_x86-0.19.8.1-7-x86_gcc2.hpkg
        /boot/system/packages/gettext_x86_libintl-0.19.8.1-7-x86_gcc2.hpkg
        /boot/system/packages/giflib-5.1.4-2-x86_gcc2.hpkg
        /boot/system/packages/giflib_x86-5.2.1-1-x86_gcc2.hpkg
        /boot/system/packages/glib2_x86-2.62.0-3-x86_gcc2.hpkg
        /boot/system/packages/glib2_x86_devel-2.62.0-3-x86_gcc2.hpkg
        /boot/system/packages/gmp_x86-6.2.1-1-x86_gcc2.hpkg
        /boot/system/packages/graphite2_x86-1.3.14-1-x86_gcc2.hpkg
        /boot/system/packages/graphite2_x86_devel-1.3.14-1-x86_gcc2.hpkg
        /boot/system/packages/grep_x86-3.6-2-x86_gcc2.hpkg
        /boot/system/packages/gutenprint8-5.3.1-1-x86_gcc2.hpkg
        /boot/system/packages/gzip-1.10-1-x86_gcc2.hpkg
        /boot/system/packages/haiku-r1~beta2_hrev54154_123-1-x86_gcc2.hpkg
        /boot/system/packages/haiku_devel-r1~beta2_hrev54154_123-1-x86_gcc2.hpkg
        /boot/system/packages/haiku_x86-r1~beta2_hrev54154_123-1-x86_gcc2.hpkg
        /boot/system/packages/harfbuzz_x86-2.7.4-2-x86_gcc2.hpkg
        /boot/system/packages/harfbuzz_x86_devel-2.7.4-2-x86_gcc2.hpkg
        /boot/system/packages/icu-57.2-3-x86_gcc2.hpkg
        /boot/system/packages/icu66_x86-66.1-2-x86_gcc2.hpkg
        /boot/system/packages/icu_x86-57.2-3-x86_gcc2.hpkg
        /boot/system/packages/jasper-2.0.16-4-x86_gcc2.hpkg
        /boot/system/packages/jasper_x86-2.0.16-4-x86_gcc2.hpkg
        /boot/system/packages/lame-3.100-3-x86_gcc2.hpkg
        /boot/system/packages/lame_x86-3.100-3-x86_gcc2.hpkg
        /boot/system/packages/lcms_x86-2.9-6-x86_gcc2.hpkg
        /boot/system/packages/libass_x86-0.15.0-1-x86_gcc2.hpkg
        /boot/system/packages/libedit-20180525_3.1-2-x86_gcc2.hpkg
        /boot/system/packages/libffi_x86-3.2.1-6-x86_gcc2.hpkg
        /boot/system/packages/libffi_x86_devel-3.2.1-6-x86_gcc2.hpkg
        /boot/system/packages/libicns-0.8.1-8-x86_gcc2.hpkg
        /boot/system/packages/libicns_x86-0.8.1-8-x86_gcc2.hpkg
        /boot/system/packages/libiconv-1.16-1-x86_gcc2.hpkg
        /boot/system/packages/libiconv_x86-1.16-1-x86_gcc2.hpkg
        /boot/system/packages/libiconv_x86_devel-1.16-1-x86_gcc2.hpkg
        /boot/system/packages/libjpeg_turbo-2.0.6-1-x86_gcc2.hpkg
        /boot/system/packages/libjpeg_turbo_x86-2.0.6-1-x86_gcc2.hpkg
        /boot/system/packages/libogg-1.3.4-1-x86_gcc2.hpkg
        /boot/system/packages/libogg_x86-1.3.4-1-x86_gcc2.hpkg
        /boot/system/packages/libopenmpt_x86-0.5.5-1-x86_gcc2.hpkg
        /boot/system/packages/libpcre_x86-8.44-1-x86_gcc2.hpkg
        /boot/system/packages/libpcre_x86_devel-8.44-1-x86_gcc2.hpkg
        /boot/system/packages/libpng16-1.6.37-1-x86_gcc2.hpkg
        /boot/system/packages/libpng16_x86-1.6.37-1-x86_gcc2.hpkg
        /boot/system/packages/libpng16_x86_devel-1.6.37-1-x86_gcc2.hpkg
        /boot/system/packages/libsolv-0.3.0_haiku_2014_12_22-3-x86_gcc2.hpkg
        /boot/system/packages/libsolv_x86-0.3.0_haiku_2014_12_22-3-x86_gcc2.hpkg
        /boot/system/packages/libtheora-1.1.1-7-x86_gcc2.hpkg
        /boot/system/packages/libtheora_x86-1.1.1-7-x86_gcc2.hpkg
        /boot/system/packages/libuv_x86-1.40.0-1-x86_gcc2.hpkg
        /boot/system/packages/libvorbis-1.3.7-1-x86_gcc2.hpkg
        /boot/system/packages/libvorbis_x86-1.3.7-1-x86_gcc2.hpkg
        /boot/system/packages/libvpx_x86-1.8.2-1-x86_gcc2.hpkg
        /boot/system/packages/libwebp-1.1.0-2-x86_gcc2.hpkg
        /boot/system/packages/libwebp_x86-1.1.0-2-x86_gcc2.hpkg
        /boot/system/packages/libxml2-2.9.10-4-x86_gcc2.hpkg
        /boot/system/packages/libxml2_x86-2.9.10-4-x86_gcc2.hpkg
        /boot/system/packages/make-4.1-4-x86_gcc2.hpkg
        /boot/system/packages/ncurses6-6.2-1-x86_gcc2.hpkg
        /boot/system/packages/ncurses6_x86-6.2-1-x86_gcc2.hpkg
        /boot/system/packages/nghttp2-1.43.0-1-x86_gcc2.hpkg
        /boot/system/packages/nghttp2_x86-1.43.0-1-x86_gcc2.hpkg
        /boot/system/packages/noto-20200106-1-any.hpkg
        /boot/system/packages/noto_sans_cjk_jp-1.004-2-any.hpkg
        /boot/system/packages/openexr-2.2.1-3-x86_gcc2.hpkg
        /boot/system/packages/openexr_x86-2.4.1-1-x86_gcc2.hpkg
        /boot/system/packages/openjpeg-2.1.2-3-x86_gcc2.hpkg
        /boot/system/packages/openjpeg_x86-2.4.0-1-x86_gcc2.hpkg
        /boot/system/packages/openssl-1.1.1i-1-x86_gcc2.hpkg
        /boot/system/packages/openssl_x86-1.1.1i-1-x86_gcc2.hpkg
        /boot/system/packages/opus-1.3.1-1-x86_gcc2.hpkg
        /boot/system/packages/opus_x86-1.3.1-1-x86_gcc2.hpkg
        /boot/system/packages/perl-5.32.0-1-x86_gcc2.hpkg
        /boot/system/packages/python3_x86-3.7.9-1-x86_gcc2.hpkg
        /boot/system/packages/readline-8.1-1-x86_gcc2.hpkg
        /boot/system/packages/readline_x86-8.1-1-x86_gcc2.hpkg
        /boot/system/packages/rhash_x86-1.4.0-2-x86_gcc2.hpkg
        /boot/system/packages/sed-4.2.1-6-x86_gcc2.hpkg
        /boot/system/packages/snappy_x86-1.1.8-1-x86_gcc2.hpkg
        /boot/system/packages/soxr_x86-0.1.3-1-x86_gcc2.hpkg
        /boot/system/packages/speex-1.2.0-4-x86_gcc2.hpkg
        /boot/system/packages/speex_x86-1.2.0-4-x86_gcc2.hpkg
        /boot/system/packages/sqlite_x86-3.34.1.0-1-x86_gcc2.hpkg
        /boot/system/packages/tar_x86-1.33-1-x86_gcc2.hpkg
        /boot/system/packages/tiff4-4.2.0-1-x86_gcc2.hpkg
        /boot/system/packages/tiff4_x86-4.2.0-1-x86_gcc2.hpkg
        /boot/system/packages/unzip-6.10c23-4-x86_gcc2.hpkg
        /boot/system/packages/wavpack_x86-5.4.0-1-x86_gcc2.hpkg
        /boot/system/packages/xz_utils_x86-5.2.5-1-x86_gcc2.hpkg
        /boot/system/packages/zlib-1.2.11-4-x86_gcc2.hpkg
        /boot/system/packages/zlib_x86-1.2.11-4-x86_gcc2.hpkg
        /boot/system/packages/zlib_x86_devel-1.2.11-4-x86_gcc2.hpkg
----- Package Info ----------------
header size:                     80
heap size:                      674
TOC size:                        41
package attributes size:        619
total size:                     754
-----------------------------------
----- Package Info ----------------
header size:                     80
heap size:                      694
TOC size:                        41
package attributes size:        661
total size:                     774
-----------------------------------
waiting for build package tidy-5.1.25-2 to be activated
waiting for build package tidy_devel-5.1.25-2 to be activated
Building ...
CMake Deprecation Warning at CMakeLists.txt:3 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- The C compiler identification is GNU 2.95
-- The CXX compiler identification is GNU 2.95
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- *** Also building DLL library SHARED, version 5.1.25, date 2015.11.21
-- *** NOTE: xsltproc NOT FOUND! Can NOT generate man page.
-- *** You need to install xsltproc in your system.
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_INSTALL_BINDIR
    CMAKE_INSTALL_DATADIR
    CMAKE_INSTALL_DATAROOTDIR
    CMAKE_INSTALL_DOCDIR
    CMAKE_INSTALL_INCLUDEDIR
    CMAKE_INSTALL_INFODIR
    CMAKE_INSTALL_LIBDIR
    CMAKE_INSTALL_LIBEXECDIR
    CMAKE_INSTALL_LOCALSTATEDIR
    CMAKE_INSTALL_MANDIR
    CMAKE_INSTALL_OLDINCLUDEDIR
    CMAKE_INSTALL_SBINDIR
    CMAKE_INSTALL_SHAREDSTATEDIR
    CMAKE_INSTALL_SYSCONFDIR


-- Build files have been written to: /sources/tidy-html5-5.1.25/build/cmake
[  1%] Building C object CMakeFiles/tidy-share.dir/src/access.c.o
cc: cannot specify -o with -c or -S and multiple compilations
CMakeFiles/tidy-share.dir/build.make:75: recipe for target 'CMakeFiles/tidy-share.dir/src/access.c.o' failed
make[2]: *** [CMakeFiles/tidy-share.dir/src/access.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[  3%] Building C object CMakeFiles/tidy-share.dir/src/istack.c.o
cc: cannot specify -o with -c or -S and multiple compilations
CMakeFiles/tidy-share.dir/build.make:103: recipe for target 'CMakeFiles/tidy-share.dir/src/istack.c.o' failed
make[2]: *** [CMakeFiles/tidy-share.dir/src/istack.c.o] Error 1
[  5%] Building C object CMakeFiles/tidy-static.dir/src/access.c.o
cc: cannot specify -o with -c or -S and multiple compilations
CMakeFiles/tidy-static.dir/build.make:75: recipe for target 'CMakeFiles/tidy-static.dir/src/access.c.o' failed
make[2]: *** [CMakeFiles/tidy-static.dir/src/access.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[  7%] Building C object CMakeFiles/tidy-share.dir/src/attrs.c.o
[  9%] Building C object CMakeFiles/tidy-static.dir/src/istack.c.o
cc: cannot specify -o with -c or -S and multiple compilations
CMakeFiles/tidy-static.dir/build.make:103: recipe for target 'CMakeFiles/tidy-static.dir/src/istack.c.o' failed
make[2]: *** [CMakeFiles/tidy-static.dir/src/istack.c.o] Error 1
[ 11%] Building C object CMakeFiles/tidy-static.dir/src/attrs.c.o
cc: cannot specify -o with -c or -S and multiple compilations
CMakeFiles/tidy-share.dir/build.make:89: recipe for target 'CMakeFiles/tidy-share.dir/src/attrs.c.o' failed
make[2]: *** [CMakeFiles/tidy-share.dir/src/attrs.c.o] Error 1
CMakeFiles/Makefile2:86: recipe for target 'CMakeFiles/tidy-share.dir/all' failed
make[1]: *** [CMakeFiles/tidy-share.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
cc: cannot specify -o with -c or -S and multiple compilations
CMakeFiles/tidy-static.dir/build.make:89: recipe for target 'CMakeFiles/tidy-static.dir/src/attrs.c.o' failed
make[2]: *** [CMakeFiles/tidy-static.dir/src/attrs.c.o] Error 1
CMakeFiles/Makefile2:112: recipe for target 'CMakeFiles/tidy-static.dir/all' failed
make[1]: *** [CMakeFiles/tidy-static.dir/all] Error 2
Makefile:155: recipe for target 'all' failed
make: *** [all] Error 2
Warning: Command '['bash', '-c', '. /wrapper-script']' returned non-zero exit status 2
Error: Build has failed - stopping.

```

**Comment by korli** [comment]
should be fixed in 90b32c1cc484a79cdf07e6d00cc270ca6912094a

**Comment by augiedoggie** [comment]
Awesome, thank you :smile:
